cirq/google/engine/engine_job.py

```diff
@@ -310,6 +314,18 @@ def _get_job_results_v1(result: v1.program_pb2.Result
                         measurements=measurements))
         return trial_results
 
+    @staticmethod
+    def _get_batch_results_v2(results: v2.batch_pb2.BatchResult
+                             ) -> List[study.TrialResult]:
+        trial_results = []
+        # Flatten to single list to match to sampler api.
+        for result in results.results:
+            sweep_results = v2.results_from_proto(result)
+            for sweep_result in sweep_results:
+                for trial_result in sweep_result:
+                    trial_results.append(trial_result)
+        return trial_results
```

Comment:
Could use the existing `_get_job_results_v2` function here:
```python
@classmethod
def _get_batch_results_v2(cls, ...):
    trial_results = []
    for result in results.results:
        trial_results.extend(cls._get_job_results_v2(result))
    return trial_results
```

Reply:
Done.